sdk/src/main/scala/com/horizen/validation/WithdrawalEpochValidator.scala

```diff
@@ -0,0 +1,60 @@
+package com.horizen.validation
+
+import com.horizen.block.SidechainBlock
+import com.horizen.params.NetworkParams
+import com.horizen.SidechainHistory
+import com.horizen.transaction.mainchain.SidechainCreation
+import com.horizen.utils.{BytesUtils, WithdrawalEpochUtils}
+import scorex.util.idToBytes
+
+import scala.util.{Failure, Success, Try}
+
+class WithdrawalEpochValidator(params: NetworkParams) extends SidechainBlockValidator {
+  override def validate(block: SidechainBlock, history: SidechainHistory): Try[Unit] = Try {
+    if (block.id.equals(params.sidechainGenesisBlockId)) {
+      // Verify that block contains only 1 MC block reference with a valid Sidechain Creation info
+      if(block.mainchainBlocks.size != 1)
+        return Failure(new IllegalArgumentException("Sidechain block validation failed for %s: genesis block should contain single MC block reference.".format(BytesUtils.toHexString(idToBytes(block.id)))))
+
+      val sidechainCreation = block.mainchainBlocks.head.sidechainRelatedAggregatedTransaction.get.mc2scTransactionsOutputs.get(0).asInstanceOf[SidechainCreation]
+      if(sidechainCreation.withdrawalEpochLength() != params.withdrawalEpochLength)
+        return Failure(new IllegalArgumentException("Sidechain block validation failed for %s: genesis block contains different withdrawal epoch length than expected in configs.".format(BytesUtils.toHexString(idToBytes(block.id)))))
+      return Success() // we don't need to check parent info for genesis block
+    }
```

Comment:
split to two functions: validation of genesis block and validation non-genesis?

Reply:
ok